src/include/hip/rpp_hip_common.hpp

```diff
@@ -135,6 +148,78 @@ inline RppStatus generate_gaussian_kernel_gpu(Rpp32f stdDev, Rpp32f* kernel, Rpp
 
 /******************** DEVICE FUNCTIONS ********************/
 
+// -------------------- Set 0 - Range checks and Range adjustment --------------------
+
+// float4 pixel check for 0-255 range
+
+__device__ __forceinline__ float4 rpp_hip_pixel_check(float4 src_f4)
+{
+    return make_float4(fminf(fmaxf(src_f4.x, 0), 255),
+                       fminf(fmaxf(src_f4.y, 0), 255),
+                       fminf(fmaxf(src_f4.z, 0), 255),
+                       fminf(fmaxf(src_f4.w, 0), 255));
+}
+
+// d_float8 adjust pixel range for different bit depths
+
+__device__ __forceinline__ void rpp_hip_adjust_range(uchar *dstPtr, d_float8 *sum_f8)
+{
```

Comment:
I need the rpp_hip_adjust_range() to perform a range adjustment only in case of the templated f16/f32/i8 variants. The U8 needs no range adjustment since its already in the correct 0-255 processing range.